src/client.ts

```diff
@@ -607,6 +622,24 @@ export class TwilioServerlessApiClient extends events.EventEmitter {
       convertApiErrorsAndThrow(err);
     }
   }
+
+  async request(
```

Comment:
Should the HTTP client be it's own class/object instead and `getClient()` then returns that said HTTP client? it would be less of a breaking change

Reply:
And it would enable people to swap in their own HTTP clients

Reply:
Yeah, I can see that working. Will look into it.

Reply:
Great :)

Reply:
We should be able to use conditional types for this. [Here's an example](https://www.typescriptlang.org/play/index.html?ssl=1&ssc=1&pln=11&pc=2#code/PTAEEsHMDsHsCcCmoAuALRBnRA6UArAV0xVVlAFsBDAa2XXE1AGNYKAHcAG2XGlQyh2XKgE9I8WIWgATAFApR7ZAHl2KcLGiYVAMwBSAZRUA5AEpZ2W7ACFYM0aAC8oAN6gkmK9sQAVJYgAXKAA5PiYWiGgAL4A3AoBoGoa1nq+iAAeKBZe1oh2Ds5uHpZ5-srBISiZKFFxCcqgABK+vgAKAIJc4FRMLiFWJFEAPqGQiLXxciCgVDIyAowNyADisCjJmtoAPL4AfEW+oDWIskxVNVEA-EnqWzq66Vk53rb2jsFHJ2eh4ZGgN02qQMxnMpR8BQ+oGgiAAboh4FNdNJmCl+EgAI6ELAoXZFMIRaAhPYACgoEzQ9mCLXaXR6mAANEIqOhgiR4HxIEzYHdrME1hteTt9gBKNxyUCSljWWA8HBcWCQMkU+zcoWYEXxaJAA)

Reply:
Here's what I meant:
```ts
// ignore these. just to make this compile in the playground
type OptionsOfJSONResponseBody = { responseType: 'json' };
type OptionsOfTextResponseBody = { responseType: 'text' };
type HTTPAlias = 'post' | 'get';

// add this
type GotOptions<T> = T extends 'text' ? OptionsOfTextResponseBody : T extends 'json' ? OptionsOfJSONResponseBody : never;

function request<T = 'json'>(method: HTTPAlias, path: string, options: GotOptions<T>) {
    console.log(method, options);
}
```

Reply:
And then you can have `request(...)` and `request<'text'>(...)`

Reply:
Solution. I ended up overloading :D i wouldn't even be mad if you leave the comments in
Imports:
```ts
import {
  HTTPAlias,
  Options,
  OptionsOfJSONResponseBody,
  OptionsOfTextResponseBody,
  Response,
} from 'got/dist/source';
```

Code:
```ts
  // request json without options
  async request(method: HTTPAlias, path: string): Promise<Response<unknown>>;
  // request json with options
  async request(
    method: HTTPAlias,
    path: string,
    options: OptionsOfJSONResponseBody
  ): Promise<Response<unknown>>;
  // request text
  async request(
    method: HTTPAlias,
    path: string,
    options: OptionsOfTextResponseBody
  ): Promise<Response<string>>;
  // general implementation
  async request(
    method: HTTPAlias,
    path: string,
    options: Options = {}
  ): Promise<unknown> {
    options.retry = RETRY_DEFAULTS;
    const gotFunction = this.client[method];
    return this.limit(() => {
      const x = gotFunction(path, options);
      return x;
    });
  }
```

Reply:
🍾